src/par/README.md

```diff
@@ -15,16 +15,363 @@ TritonPart is an open-source constraints-driven partitioner.  It can be used to
   - Embedding-aware partitioning
   
 ## Dependency
+
 We use Google OR-Tools as our ILP solver.  Please install Google OR-Tools following the [instructions](https://developers.google.com/optimization/install).
 
-## How to partition a hypergraph in the way you would using hMETIS (min-cut partitioning)
-``` shell
+```{warning}
+Due to a build issue, TritonPart is not supported for MacOS. Stay tuned to this page for updates!
+```
+
+## Algorithm at a glance
+
+An overview of the TritonPart algorithm is shown below. It takes as inputs 
+- Hypergraph $H(V,E)$ in `.hgr` format
+- Number of supervision iterations $\beta$
+- The number of blocks in a partitioning solutions $K$
+- An initial partitioning solution $S_{init}$
+- The allowed imbalance between blocks $\epsilon$
+
+The output is an improved partitioning solution $S_{out}$. 
+
+There are 3 main steps, mainly 1) vertex embedding generation, 2) tree construction and partitioning, 3) cut-overlay clustering and ILP-based Partitioning and 4) solution refinement. We will go through each step in turn.
+
+1. Vertex Embedding Generation
+- An initial vertex embedding with solution $S_i$ is generated as a hint.
+- For bipartitioning ($K=2$), we solve a generalized eigenvalue problem to obtain the first $m$ nontrivial eigenvectors
+$X \in \mathcal{R}^{|V|\times m}$
+- For multi-way partitioning ($K>2$), we decompose the $K$-way solution into $K$ bipartitioning solutions. Then these will be solved together in a generalized eigenvalue problem, from which we obtain the first $m$ nontrivial eigenvector per solution. From this, we obtain $K \times m $ eigenvectors. Dimension reduction is then performed using Linear Discriminant Analysis (LDA).
+
+2. Tree Construction and Partitioning
+- The vertex embedding generated by Step 1 is used to compute a family of trees that each *distills* 
+the cut structure of the hypergraph.
+- Subsequently, efficiently tree-based algorithms are employed to obtain partitioning solutions $\{S_{T_0},S_{T_1, ...}\}$
+- All tree partitioning solutions are further refined using the multi-way Fiduccia-Mattheyes (FM) algorithm. 
+
+3. Cut-Overlay Clustering and Integer Linear Programming (ILP) based Partitioning
+- Algorithm picks the top $\sigma$ solutions from Step 2, and overlays these solution to generate a clustered hypergraph.
+- Then it generates a solution $S_i$ by performing ILP-based partitioning.
+- Solution $S_i$ is further leveraged as a hint to drive the next iteration. 
+
+4. Solution Refinement
+- After obtaining the candidate solutions from Steps 1-3, the algorithm generates the improved solution $S_out$ 
+by (i) performing cut-overlay clustering and ILP-based partitioning on candidate solutions and (ii) refining the solution
+with the FM algorithm. 
+
+
+![TritonPart overview](./doc/kspecpart.png)
```

Comment:
Can we please make this an SVG instead of PNG?

Reply:
Updated